packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -79,14 +75,25 @@ class HotRunner extends ResidentRunner {
              packagesFilePath: packagesFilePath,
              saveCompilationTrace: saveCompilationTrace,
              stayResident: stayResident,
-             ipv6: ipv6);
+             ipv6: ipv6) {
+                watcher = DirectoryWatcher(projectRootPath ?? fs.currentDirectory.path);
```

Comment:
I thought we talked about how this will regress dependency analysis: files outside of current directory won't be watched, unused files inside of current directory will be reported to the compiler.
The right solution is to have incremental compiler report files that comprise the application(similar to how list of files produced for .dep-file used for gradle), watch those.

Reply:
For most projects the current directory contains the only sources that will be changed. For instance, on a pub based workflow to update sources I need to kill the runner and run pub get too.

How much execution time are we willing to trade to make dependency tracking more accurate? The changes here represent ~0.5 seconds across all hot reloads on windows for example. Would requiring a hot restart to update other directories be a reasonable compromise?

Reply:
Querying the compiler for the list of sources it used should be instantaneous - it already has this information because it just compiled all of them.

Reply:
Understood. I think we both agree that the analyzer based approach is flawed - tomorrow lets discuss how we can proceed on incorporating the incremental compiler reports.